src/main/java/com/nutomic/syncthingandroid/activities/DeviceActivity.java

```diff
@@ -207,6 +250,24 @@ public void onPause() {
     public void onSaveInstanceState(Bundle outState) {
         super.onSaveInstanceState(outState);
         outState.putString("device", new Gson().toJson(mDevice));
+        if (mIsCreateMode){
+            outState.putBoolean(IS_SHOWING_DISCARD_DIALOG, mIsShowingDiscardDialog);
+            if(mDiscardDialog != null){
+                mDiscardDialog.cancel();
+            }
```

Comment:
Is there any reason you cancel the dialogs here? I think it might be possible that `onSaveInstanceState()` is called, but the activity is not killed (not sure from the documentation). Either way, rotating the screen cancels all existing dialogs, so there should be no need to cancel them manually.